### ISSUE TITLE
aot_export_joint_simple on plain callable (not graph module) doesn't attach stack traces

### ISSUE THREAD

**Issue by ezyang** [pytorch/pytorch#102205]
### 🐛 Describe the bug

Repro example:

```
import torch
from torch._functorch.aot_autograd import aot_export_joint_simple
    
def f(x, weight):
    def cb(grad_out):
        y = grad_out.sin()  # stand-in for collective result
        weight.data.copy_(y)
        # Better would be:
        #   weight.data = y
        return grad_out
    r = x @ weight
    r.register_hook(cb) 
    return (r,)

gm = aot_export_joint_simple(
    f,
    [torch.randn(2, 3, requires_grad=True), torch.randn(3, 4)],
    trace_joint=True,
)
gm.print_readable()
```

gives

```
        # No stacktrace found for following nodes
```

It would be nice to get the native stack traces in this case.

### Versions

main

cc @bhosmer @smessmer @ljk53 @bdhirsh @msaroufim @wconstab @anijain2305 @zou3519 @ngimel

**Comment by bdhirsh** [comment]
the example in the repro seems to error for me, but this repro seems to work (and print no stack traces in the graph)
```
import torch
from torch._functorch.aot_autograd import aot_export_joint_simple

def f(x, weight):
    r = x @ weight
    y = r.sin()
    return (y,)

gm = aot_export_joint_simple(
    f,
    [torch.randn(4, 4, requires_grad=True), torch.randn(4, 4)],
    trace_joint=True,
)
gm.print_readable()
```

which prints:
```
class joint_helper(torch.nn.Module):
    def forward(self, primals, tangents):
        primals_1: f32[4, 4], primals_2: f32[4, 4], tangents_1: f32[4, 4], = fx_pytree.tree_flatten_spec([primals, tangents], self._in_spec)
        # No stacktrace found for following nodes
        mm: f32[4, 4] = torch.ops.aten.mm.default(primals_1, primals_2);  primals_1 = None
        sin: f32[4, 4] = torch.ops.aten.sin.default(mm)
        cos: f32[4, 4] = torch.ops.aten.cos.default(mm);  mm = None
        mul: f32[4, 4] = torch.ops.aten.mul.Tensor(tangents_1, cos);  tangents_1 = cos = None
        t: f32[4, 4] = torch.ops.aten.t.default(primals_2);  primals_2 = None
        mm_1: f32[4, 4] = torch.ops.aten.mm.default(mul, t);  mul = t = None
        return pytree.tree_unflatten([sin, mm_1, None], self._out_spec)
```

**Comment by bdhirsh** [comment]
oh @tugsbayasgalan did you end up fixing this? (If not / you don't plan on working on it soon feel free to unassign)

**Comment by tugsbayasgalan** [comment]
Will take a look again, sorry for the delay

**Comment by keyan** [comment]
Hey @tugsbayasgalan - looks like this got lost in the shuffle so I'm happy to work on it if you'd like.

I opened a draft PR and just need a pointer to get it finalized. Thanks!
